core/src/HIP/Kokkos_HIP_Team.hpp

```diff
@@ -170,6 +170,9 @@ class HIPTeamMember {
       ::Kokkos::Experimental::Impl::in_place_shfl(
           val, tmp, blockDim.x * thread_id, blockDim.x * blockDim.y);
     }
+    // FIXME_HIP the synchronization point is necessary because of a race
+    // condition in shuffle
+    __syncthreads();
```

Comment:
Why did you not just put the synchronization point in the shuffle function?

Reply:
again this doesn't all doesn't work since these functions are not block synchronous called ...